### PR TITLE
Convert Filebeat icinga.* to ECS

### PR DESCRIPTION
**Pull request by webmat**
Caveats

- ECS doesn't have support for "facility" yet. Not sure if we should mix Icinga and Syslog facilities in any case. So haven't touched this field
- This module doesn't do deep parsing of the Icinga messages. There are interesting bits (e.g. PID, execution_time) that may eventually be interesting to extract out. This is out of scope for this initial mapping to ECS, however.
- Not copying original message to log.original, as this is out of scope for this mapping. Will be addressed with an optional toggle at the Beats level.

Renames

- icinga.debug.message => message
- icinga.main.message => message
- icinga.startup.message => message
- icinga.debug.severity => log.level
- icinga.main.severity => log.level
- icinga.startup.severity => log.level

TODO

- [x] Commit missing updates to the `-expected.json` file
- [x] Alias renamed fields to their ECS counterpart
- [x] Document field migrations in ecs-migration.yml
- [x] Changelog

### CODE REVIEW

**Comment by webmat** [comment]
jenkins, test this bro

**Comment by ruflin** [comment]
Caveats: Agree to "ignore" all of them for now.

**Comment by webmat** [comment]
jenkins, test this

File: dev-tools/ecs-migration.yml [hunk 1]
```diff
@@ -447,12 +448,39 @@
   to: message
   alias: true
 
-# From Auditbeat's auditd module.
+## Icinga module
+
+- from: icinga.debug.message
+  to: message
+  alias: true
+- from: icinga.debug.severity
```
[the remark this answers is not in the document]
Comment (webmat) [reply dev-tools/ecs-migration.yml]: Haha oops! Copy pasta

**Comment by webmat** [comment]
Timeouts on ML tests. Will restart the filebeat tests, should be unrelated to my very simple changes.

```
Test modules list command ... ok (0.0913s)
======================================================================
ERROR: Test ML are installed in all possible ways [with setup_flag=True, modules_flag=False]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/filebeat/build/python-env/local/lib/python2.7/site-packages/parameterized/parameterized.py", line 392, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_ml.py", line 46, in test_ml_setup
    self._run_ml_test(setup_flag, modules_flag)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_ml.py", line 136, in _run_ml_test
    max_timeout=60)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 363, in wait_log_contains
    name=name)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 340, in wait_until
    "Waited {} seconds.".format(max_timeout))
TimeoutError: Timeout waiting for 'log_contains' to be true. Waited 60 seconds.
-------------------- >> begin captured stdout << ---------------------
Using elasticsearch: http://elasticsearch:9200
Test setup_flag: True, modules_flag: False
--------------------- >> end captured stdout << ----------------------
======================================================================
ERROR: Test ML are installed in all possible ways [with setup_flag=True, modules_flag=True]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/filebeat/build/python-env/local/lib/python2.7/site-packages/parameterized/parameterized.py", line 392, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_ml.py", line 46, in test_ml_setup
    self._run_ml_test(setup_flag, modules_flag)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_ml.py", line 136, in _run_ml_test
    max_timeout=60)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 363, in wait_log_contains
    name=name)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 340, in wait_until
    "Waited {} seconds.".format(max_timeout))
TimeoutError: Timeout waiting for 'log_contains' to be true. Waited 60 seconds.
-------------------- >> begin captured stdout << ---------------------
Using elasticsearch: http://elasticsearch:9200
License already enabled
Test setup_flag: True, modules_flag: True
--------------------- >> end captured stdout << ----------------------
======================================================================
ERROR: Test ML are installed in all possible ways [with setup_flag=False, modules_flag=False]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/filebeat/build/python-env/local/lib/python2.7/site-packages/parameterized/parameterized.py", line 392, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_ml.py", line 46, in test_ml_setup
    self._run_ml_test(setup_flag, modules_flag)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_ml.py", line 136, in _run_ml_test
    max_timeout=60)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 363, in wait_log_contains
    name=name)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 340, in wait_until
    "Waited {} seconds.".format(max_timeout))
TimeoutError: Timeout waiting for 'log_contains' to be true. Waited 60 seconds.
-------------------- >> begin captured stdout << ---------------------
Using elasticsearch: http://elasticsearch:9200
License already enabled
Test setup_flag: False, modules_flag: False
--------------------- >> end captured stdout << ----------------------
======================================================================
ERROR: Test ML are installed in all possible ways [with setup_flag=False, modules_flag=True]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/filebeat/build/python-env/local/lib/python2.7/site-packages/parameterized/parameterized.py", line 392, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_ml.py", line 46, in test_ml_setup
    self._run_ml_test(setup_flag, modules_flag)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_ml.py", line 136, in _run_ml_test
    max_timeout=60)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 363, in wait_log_contains
    name=name)
  File "/go/src/github.com/elastic/beats/filebeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 340, in wait_until
    "Waited {} seconds.".format(max_timeout))
TimeoutError: Timeout waiting for 'log_contains' to be true. Waited 60 seconds.
-------------------- >> begin captured stdout << ---------------------
Using elasticsearch: http://elasticsearch:9200
License already enabled
Test setup_flag: False, modules_flag: True
--------------------- >> end captured stdout << ----------------------
----------------------------------------------------------------------
XML: /go/src/github.com/elastic/beats/filebeat/build/TEST-system.xml
[error] 15.34% test_ml.Test.test_ml_setup_0: 92.3972s
[success] 11.59% test_autodiscover.TestAutodiscover.test_docker: 69.8451s
[error] 9.97% test_ml.Test.test_ml_setup_1: 60.0760s
[error] 9.97% test_ml.Test.test_ml_setup_2: 60.0727s
[error] 9.97% test_ml.Test.test_ml_setup_3: 60.0460s
[success] 1.56% test_registrar.Test.test_restart_state: 9.4150s
```

**Comment by ruflin** [comment]
It seems something changed on master in ML, ignore them for now: https://github.com/elastic/beats/pull/9718

**Comment by webmat** [comment]
@ruflin I'll wait for this fix to the ML tests to be in and rebase, since my PRs are specifically Filebeat modules. I want the full test run.

However can you give the changes one last look and approve the PR?

File: filebeat/module/icinga/main/test/test.log-expected.json [hunk 2]
```diff
@@ -4,32 +4,32 @@
         "event.dataset": "main", 
         "event.module": "icinga", 
         "icinga.main.facility": "Notification", 
-        "icinga.main.message": "Sending 'Recovery' notification 'demo!load!mail-icingaadmin for user 'on-call'", 
-        "icinga.main.severity": "information", 
         "input.type": "log", 
-        "log.offset": 0
+        "log.level": "information",
```
Comment (ruflin) [inline filebeat/module/icinga/main/test/test.log-expected.json]: we should probably also standardise these in ECS so it's always "INFO". Nothing todo here, just a general comment.

**Pull request merged**